### ISSUE TITLE
Should look in $XDG_DATA_DIRS/icons for icons

### ISSUE THREAD

**Issue by sgraf812** [sagb/alttab#82]
The manpage mentions that `alttab` will look in these locations for icon files:

```
              /usr/share/icons
              /usr/local/share/icons
              ~/.icons
              ~/.local/share/icons
```

With reason being that it complies to the freedesktop standard. But the same [standard](https://standards.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html) says that we should look under $XDG_DATA_DIRS/icons instead of ~/.local/share/icons. I'd like a fallback mechanism for `alttab`, not unlike `${XDG_DATA_DIRS:-~/.local/share}/icons` in bash.

**Comment by sgraf812** [comment]
I found this comment:

https://github.com/sagb/alttab/blob/1df30b15f8ec479937f80437c737cfdc6ce0d469/src/icon.c#L99-L108

I wonder why this deliberately ignores $XDG_DATA_DIR?

**Comment by sagb** [comment]
I didn't use XDG vars for the reason you realised when trying #83: it brings complication.
But, well, let's do this.
Note that my implementation still violates standard (deliberately): XDG variable isn't used "instead of" hardcoded directories, but along with them.

**Comment by sagb** [comment]
Feel free to reopen.

**Comment by sgraf812** [comment]
I think the best way forward would be to copy a function like https://github.com/LiquidFM/libxdg/blob/ffc4f1b81a176b995c10855ba8a60ff39e71c6a5/src/basedirectory/xdgbasedirectory.h#L60. Alternatively, we could depend on libxdg-basedir, which seems widely available, and call https://github.com/devnev/libxdg-basedir/blob/86ea74ee174c6bee204729293b2b292492d35cb3/include/basedir.h#L106, and do some of the logic of libxdg ourselves. I think I'd favor the former; the license (LGPL and Academic Free License) seem to be compatible.

**Comment by sgraf812** [comment]
Also Re: XDG-compliance: May I ask why you don't want to follow the standard completely? That would make the implementation a little easier, I think.

**Comment by sgraf812** [comment]
There's even this function in libxdg: https://github.com/LiquidFM/libxdg/blob/ffc4f1b81a176b995c10855ba8a60ff39e71c6a5/src/themes/xdgtheme.h#L58 I think this should do everything we want it to without even having to think about paths etc.

XDG-compliant lookup is also [more complicated than it seems](https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html), because of the interaction with theme inheritance, icon sizes and contexts. This function seems to neatly abstract that.

**Comment by sagb** [comment]
The target audience of alttab are probably people which use minimalistic or customized environments, where chances are high that XDG variables are unset or random.
Actually, during the years of existence of these variables, I fight with them, because if you don't look for them, then Desktop, Downloads directories and other unwanted crap are suddenly born everywhere. So conformance with _these_ standards are not important for me. I wanted alttab to depend only on well-recognized code and standards like xlib and libpng.

**Comment by sgraf812** [comment]
I'm starting to understand what you mean. A shame that they don't provide a proper portable C implementation of their standard.

Anyway, I'm using Nix, where there actually aren't any icons under ~/.local/share, but under ~/.nix-profile/share. Would you be open to a PR that adds an command line option/recognises an env var that adds to the searched icon paths?

**Comment by sagb** [comment]
So use XDG_DATA_DIRS? Why don't you like limited support for it added in ea5ad54 ?

**Comment by sgraf812** [comment]
Oh, I didn't read the closed notification carefully. Thanks for implementing a fix! I tried it and ran into a segfault caused by `icon_dirs` not being initialised. Pushed a fix in #84.
